react/components/SearchBar/components/AutocompleteInput.js

```diff
@@ -26,17 +27,21 @@ export default class AutocompleteInput extends Component {
   }
 
   render() {
-    const { onGoToSearchPage, onClearInput, compactMode, value, ...restProps } = this.props
+    const { onGoToSearchPage, onClearInput, compactMode, value, hasIconLeft, iconClasses, ...restProps } = this.props
 
     const suffix = (
       <span className="flex items-center pointer" onClick={value ? onClearInput : undefined} >
         {value
           ? <IconClose className="pa0" size={20} />
-          : <IconSearch />
+          : !hasIconLeft && <Icon id="hpa-search" className={iconClasses} />
```

Comment:
I think is better to use a getter here instead of using a ternary operator and then an expression. Something like

```javascript
get myIcon() {
  if (value) {
    return <IconClose ... />
  } else {
    return !hasIconLeft && ...
  }
}
```

Reply:
About the name of the getter I don't know a good one, in `product-summary` we was doing some getters like `get renderProductName()`, `get renderProductPrice()` and so on. But they all return the same component, `get renderIcon()` maybe a good name.